File: 프로그래머스/LEVEL2/Q42626.java
package programmers;

// Q42626 더 맵게
// https://school.programmers.co.kr/learn/courses/30/lessons/42626?language=java
import java.util.PriorityQueue;

class Solution {
	public int solution(int[] scoville, int k) {
		PriorityQueue<Integer> pq = new PriorityQueue<>();
		int result = 0;
		for (int i : scoville) {
			pq.offer(i);
		}
		while (pq.peek() < k) {
			Integer a = pq.poll();
			Integer b = pq.poll();
			if (a == null || a <= 0 || b == null) {
				return -1;
			}
			pq.offer(a + b * 2);
			result++;
		}
		return result;
	}
}

File: 프로그래머스/LEVEL2/Q42626.py
# Q42626 더 맵게
# https://school.programmers.co.kr/learn/courses/30/lessons/42626?language=python3
import heapq

def solution(scoville:list, K):
    heapq.heapify(scoville)
    result = 0
    while scoville[0] < K and len(scoville) > 1:
        curr = heapq.heappop(scoville) + heapq.heappop(scoville) * 2
        heapq.heappush(scoville, curr)
        result+=1
    if scoville[0] < K:
        return -1
    else:
        return result
